packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -310,19 +360,29 @@ class _HeroState extends State<Hero> {
       'A Hero widget cannot be the descendant of another Hero widget.'
     );
 
-    if (_placeholderSize != null) {
-      if (widget.placeholderBuilder == null) {
-        return SizedBox(
-          width: _placeholderSize.width,
-          height: _placeholderSize.height,
-        );
-      } else {
-        return widget.placeholderBuilder(context, widget.child);
-      }
+    final bool isHeroInFlight = _placeholderSize != null;
+
+    if (isHeroInFlight && widget.placeholderBuilder != null) {
+      return widget.placeholderBuilder(context, _placeholderSize, widget.child);
     }
-    return KeyedSubtree(
-      key: _key,
-      child: widget.child,
+
+    if (isHeroInFlight && !_shouldIncludeChild) {
+      return SizedBox(
+        width: _placeholderSize.width,
+        height: _placeholderSize.height,
+      );
+    }
+
+    return SizedBox(
+      width: _placeholderSize?.width,
```

Comment:
This breaks getting sematnics for a FloatingActionButton, e.g.

```
  await tester.pumpWidget(MaterialApp(
        home: FloatingActionButton(
      onPressed: () {},
    )));

    expect(
        tester.getSemantics(find.byType(FloatingActionButton)),
        matchesSemantics(
          hasTapAction: true,
          hasEnabledState: true,
          isButton: true,
          isEnabled: true,
        ));
 
```

Now returns the semantics node for the parent of the widget rather than the semantics node for the widget.

Reply:
Fix for this is in https://github.com/flutter/flutter/pull/34512